contracts/atomic-swap/src/state.rs

```diff
@@ -0,0 +1,92 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::{
+    CanonicalAddr, Coin, Env, Order, ReadonlyStorage, StdError, StdResult, Storage,
+};
+use cosmwasm_storage::{bucket, bucket_read, prefixed_read, Bucket, ReadonlyBucket};
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug, Default)]
+pub struct AtomicSwap {
+    /// This is hex-encoded sha-256 hash of the preimage (must be 32*2 = 64 chars)
+    pub hash: String,
+    pub recipient: CanonicalAddr,
+    pub source: CanonicalAddr,
+    pub end_height: u64,
+    pub end_time: u64,
+    /// Balance in native tokens
+    pub balance: Vec<Coin>,
+}
+
+impl AtomicSwap {
```

Comment:
I would use [`cw0::Expiration`](https://github.com/CosmWasm/cosmwasm-plus/blob/master/packages/cw0/src/expiration.rs#L6-L15) instead of `end_height` and `end_time`. We have the logic elsewhere and it standardized this input field